Tree

kmuel4 opened the pull request:
made progress on creating constructor for Tree

kmuel4 committed:
Update Tree.java

kmuel4 committed:
Update Tree.java

kmuel4 committed:
Create TreeView

kmuel4 committed:
Create TreeView

kmuel4 committed:
Update Tree.java

kmuel4 committed:
Update Tree.java

nmoon1 commented on model/Tree.java:
```diff
@@ -1,5 +1,38 @@
+import userinterface.View;
+import userinterface.ViewFactory;
+
+public class Tree extends TLC implements IView, IModel  {
```
should extend EntityBase rather than TLC only needs to implement IModel:
public class Tree extends EntityBase implements Imodel {...}

kmuel4 committed:
Delete TreeView

pushed by mistake

kmuel4 committed:
Delete TreeView

pushed by mistake

kmuel4 committed:
Update Tree.java

kmuel4 committed:
Update Tree.java

kmuel4 committed:
Merge branch 'Tree' of https://github.com/nmoon1/CSC429ScoutProject into Tree

kmuel4 committed:
finished rough draft Tree.java

kmuel4 committed:
finished rough draft Tree.java

nmoon1 commented on model/Tree.java:
```diff
@@ -1,5 +1,257 @@
+
+	// Constructor for existing tree
+	//---------------------------------------------------------------------
+	public Tree(int Barcode) throws InvalidPrimaryKeyException
```
barcode will be a string

nmoon1 commented on model/Tree.java:
```diff
@@ -1,5 +1,257 @@
+				"; Status: " + persistentState.getProperty("Status") +
+				"; DateStatusUpdated: " + persistentState.getProperty("DateStatusUpdated") + "\n";
+	}
+	public void createAndShowTreeView()
```
views should be handled by the TLC not here, whenever you need to change views you should call the stateChangeRequest method of the TLC and add an entry to that method to handle the request, I can show you in class what I mean if you are confused

nmoon1 commented on model/Tree.java:
```diff
@@ -1,5 +1,257 @@
+		swapToView(currentScene);
+	}
+
+	public String getTreeBarcode()
```
you don't need getters, the getState method handles this

kmuel4 committed:
Merge branch 'Tree' of https://github.com/nmoon1/CSC429ScoutProject into Tree

kmuel4 committed:
Merge branch 'main' into Tree

kmuel4 committed:
updated .gitignore

made it so .idea is not included in changes

kmuel4 committed:
Merge branch 'Tree' of https://github.com/nmoon1/CSC429ScoutProject into Tree

kmuel4 committed:
updated Tree.java and added AddTreeView.java

updated Tree.java and added AddTreeView.java, changed .gitignore to ignore .idea
